ueberdosis/tiptap
# fix: streamline html package generation functions to work as expected

## Changes Overview
This PR removes logic that was specific to the `@tiptap/html` package and didn't work the same way the core `generateHTML` and `generateJSON` functions work. This caused issues with a few extensions (for example the Youtube extension not being able to be generated back to HTML from JSON, see #4089)

I kept the original HTML `getJSON` and `getHTML` functions to not break existing code for users but maybe we should add a deprecation message to it?

## Implementation Approach
I removed the logic of the functions of the html package and just used a reference to the core packages util functions.

## Testing Done
I tried running JSON and HTML generation steps before and after and checked if unexpected things were happening or the output was false.

## Verification Steps
1. Try to check out this PR
2. Create a demo and simply run the code example below
3. Verify output

```js
import { generateJSON, generateHTML } from '@tiptap/html'
const extensions = [Document, Paragraph, Text, Youtube]
const content `<p>Hello</p><div data-youtube-video><iframe src="..."></iframe></div>`

const json = generateJSON(content, extensions)
const html = generateHTML(json, extensions)

console.log(json, html)
```

## Additional Notes


## Checklist
- [x] I have renamed my PR according to the naming conventions. (e.g. `feat: Implement new feature` or `chore(deps): Update dependencies`)

## Related Issues
- fixes #4089
